source/extensions/stat_sinks/common/statsd/statsd.h

```diff
@@ -65,6 +70,8 @@ class UdpStatsdSink : public Stats::Sink {
   ThreadLocal::SlotPtr tls_;
   Network::Address::InstanceConstSharedPtr server_address_;
   const bool use_tag_;
+  // Prefix for all flushed stats.
+  const std::string& prefix_;
```

Comment:
Is there a particular reason that this needs to be a reference?  It's usually assumed that when a const reference is passed in (via the constructor above, for instance) that it is only guaranteed to live for the duration of that function call - meaning it shouldn't be stored and used later (same for TCP).

Reply:
You're right. This shouldn't be a reference. I've changed this to `const std::string` The `getPrefix()` methods for both Udp & Tcp sinks have been changed to return `const std::string`